### ISSUE TITLE
False positive DAR301 for optional keyword parameter when using Numpy-style docstrings

### ISSUE THREAD

**Issue by amv213** [terrencepreilly/darglint#168]
```python
def foo(x: str = 'bar') -> None:
    """[summary]

    Parameters
    ----------
    x : str, optional
        [description], by default 'bar'
    """

    print(x)
```

raises

`DAR103 Parameter type mismatch:  ~x: expected str but was str, optional`

when `<type>, optional`  is the [numpydoc ](https://numpydoc.readthedocs.io/en/latest/format.html#parameters) way of indicating that the given parameter is optional and has a default value. Darglint might be interpreting it as corresponding to a `typing.Optional` type definition - which in this case is not actually appropriate.

**Comment by terrencepreilly** [comment]
This has been brought up before, many times. See issues #123, #114, #112, and #95.
